examples/window-app/silabs/efr32/include/WindowManager.h

```diff
@@ -0,0 +1,251 @@
+        CoverTypeChange,
+        TiltModeChange,
+
+        // Cover Attribute update events
+        AttributeChange,
+
+        // Provisioning events
+        ProvisionedStateChanged,
+        ConnectivityStateChanged,
+        BLEConnectionsChanged,
+        WinkOff,
+        WinkOn,
+    };
+
+    struct Event
+    {
+        Event(EventId id) : mId(id), mEndpoint(0) {}
+        Event(EventId id, chip::EndpointId endpoint) : mId(id), mEndpoint(endpoint) {}
+        Event(EventId id, chip::EndpointId endpoint, chip::AttributeId attributeId) :
+            mId(id), mEndpoint(endpoint), mAttributeId(attributeId)
+        {}
+
+        EventId mId;
+        chip::EndpointId mEndpoint;
+        chip::AttributeId mAttributeId;
+    };
+
+    struct Cover
+    {
+        void Init(chip::EndpointId endpoint);
+        void Finish();
+
+        void LiftUpdate(bool newTarget);
+        void LiftGoToTarget() { LiftUpdate(true); }
+        void LiftContinueToTarget() { LiftUpdate(false); }
+        void LiftStepToward(OperationalState direction);
+        void LiftSchedulePositionSet(chip::Percent100ths position) { SchedulePositionSet(position, false); }
+        void LiftScheduleOperationalStateSet(OperationalState opState) { ScheduleOperationalStateSet(opState, false); }
+
+        void TiltUpdate(bool newTarget);
+        void TiltGoToTarget() { TiltUpdate(true); }
+        void TiltContinueToTarget() { TiltUpdate(false); }
+        void TiltStepToward(OperationalState direction);
+        void TiltSchedulePositionSet(chip::Percent100ths position) { SchedulePositionSet(position, true); }
+        void TiltScheduleOperationalStateSet(OperationalState opState) { ScheduleOperationalStateSet(opState, true); }
+
+        void UpdateTargetPosition(OperationalState direction, bool isTilt);
+        void StepToward(OperationalState direction, bool isTilt);
+
+        Type CycleType();
+
+        static void OnLiftTimeout(Timer & timer);
+        static void OnTiltTimeout(Timer & timer);
+
+        chip::EndpointId mEndpoint = 0;
+
+        Timer * mLiftTimer            = nullptr;
+        Timer * mTiltTimer            = nullptr;
+        OperationalState mLiftOpState = OperationalState::Stall;
+        OperationalState mTiltOpState = OperationalState::Stall;
+
+        struct CoverWorkData
+        {
+            chip::EndpointId mEndpointId;
+            bool isTilt;
+
+            union
+            {
+                chip::Percent100ths percent100ths;
+                OperationalState opState;
+            };
+        };
+
+        void SchedulePositionSet(chip::Percent100ths position, bool isTilt);
+        static void CallbackPositionSet(intptr_t arg);
+        void ScheduleOperationalStateSet(OperationalState opState, bool isTilt);
+        static void CallbackOperationalStateSet(intptr_t arg);
+    };
+
+    static WindowManager & Instance();
+
+    WindowManager();
+
+    CHIP_ERROR Init();
+    CHIP_ERROR Start();
+    CHIP_ERROR Run();
```

Comment:
Should cleanup all of this with the Start, Run, Stop and finish

Reply:
Will merge Init and Start into a single Init function. Removed Finish function since we don't care about memory leaks in our use case.